### PR TITLE
Merkle-tree / milestone-verification

### PR DESCRIPTION
**Pull request by muXxer**
# Description

I added a merkle tree package and made some changes to verify milestones.

## Type of change

- [X] Enhancement (non-breaking change which adds functionality)

# How Has This Been Tested?

- [X] Added test for merkle tree
- [X] Added test for milestone verification

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have written example code according to the contribution guidelines
- [X] I have added tests using ginkgo that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by muXxer** [comment]
:wen: ? :carlos:

File: merkle/merkle.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+			copy(tree[parentIdx:parentIdx+HashTrinarySize], trits)
+			h.Reset()
+		}
+		curSize = (curSize + 1) >> 1
+	}
+
+	return tree, nil
+}
+
+// MerkleBranch creates the merkle branch to generate back root from index
```
Comment (luca-moser) [inline merkle/merkle.go]:
```suggestion
// MerkleBranch creates the merkle branch to generate back root from index.
```

File: merkle/merkle.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+			copy(siblings[i*HashTrinarySize:(i+1)*HashTrinarySize], tree[siteIndex:siteIndex+HashTrinarySize])
+		}
+
+		siblingIndex >>= 1
+		i++
+	}
+	return siblings, nil
+}
+
+// MerkleRoot generates a merkle root from a hash and his siblings
```
Comment (luca-moser) [inline merkle/merkle.go]:
```suggestion
// MerkleRoot generates a merkle root from a hash and his siblings,
```

File: signing/signing_milestone_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+	Context("MerkleRoot()", func() {
+		It("Check milestone signature", func() {
```
Comment (luca-moser) [inline signing/signing_milestone_test.go]:
```suggestion
		It("Checks milestone signature", func() {
```

File: merkle/merkle.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
+// Package merkle provides functions for creating and validating merkle trees
```
Comment (luca-moser) [inline merkle/merkle.go]:
```suggestion
// Package merkle provides functions for creating and validating merkle trees.
```

**Pull request merged**